pyk/src/pyk/kore/rpc.py

```diff
@@ -304,10 +308,12 @@ def request(self, method: str, **params: Any) -> dict[str, Any]:
         server_addr = self._transport.description()
         _LOGGER.info(f'Sending request to {server_addr}: {old_id} - {method}')
         req = json.dumps(payload)
+        base_name = self._bug_report_id if self._bug_report_id is not None else 'kore_rpc'
+        req_name = f'{base_name}/{self.client_id}/{old_id:03}'
         if self._bug_report:
-            bug_report_request = f'rpc_{self._bug_report_id}/{old_id:03}_request.json'
+            bug_report_request = f'{req_name}_request.json'
             self._bug_report.add_file_contents(req, Path(bug_report_request))
-            self._bug_report.add_command(self._transport.command(self._bug_report_id, old_id, bug_report_request))
+            self._bug_report.add_command(self._transport.command(req_name, bug_report_request))
```

Comment:
I wonder, is it possible to pull all this logic into `Transport` (and its subclasses)? Then `command` and `client_id`  can become private on `Transport`.